http-server-netty/src/main/java/io/micronaut/http/server/netty/handler/accesslog/element/AccessLogFormatParser.java

```diff
@@ -0,0 +1,359 @@
+ * <li><b>%a</b> - Remote IP address</li>
+ * <li><b>%A</b> - Local IP address</li>
+ * <li><b>%b</b> - Bytes sent, excluding HTTP headers, or '-' if no bytes were sent</li>
+ * <li><b>%B</b> - Bytes sent, excluding HTTP headers</li>
+ * <li><b>%h</b> - Remote host name</li>
+ * <li><b>%H</b> - Request protocol</li>
+ * <li><b>%{<header>}i</b> - Request header. If the argument is omitted (<b>%i</b>) all headers will be printed</li>
+ * <li><b>%{<header>}o</b> - Response header. If the argument is omitted (<b>%o</b>) all headers will be printed</li>
+ * <li><b>%{<cookie>}C</b> - Request cookie (COOKIE). If the argument is omitted (<b>%C</b>) all cookies will be printed</li>
+ * <li><b>%{<cookie>}c</b> - Response cookie (SET_COOKIE). If the argument is omitted (<b>%c</b>) all cookies will be printed</li>
+ * <li><b>%l</b> - Remote logical username from identd (always returns '-')</li>
+ * <li><b>%m</b> - Request method</li>
+ * <li><b>%p</b> - Local port</li>
+ * <li><b>%q</b> - Query string (excluding the '?' character)</li>
+ * <li><b>%r</b> - First line of the request</li>
+ * <li><b>%s</b> - HTTP status code of the response</li>
+ * <li><b>%{<format>}t</b> - Date and time. If the argument is ommitted the Common Log Format format is used ("'['dd/MMM/yyyy:HH:mm:ss Z']'").
+ * If the format starts with begin: (default) the time is taken at the beginning of the request processing. If it starts with end: it is the time when the log entry gets written, close to the end of the request processing.
+ * The format should follow the DateTimeFormatter syntax.</li>
+ * <li><b>%u</b> - Remote user that was authenticated. Not implemented. Prints '-'.</li>
+ * <li><b>%U</b> - Requested URI</li>
+ * <li><b>%v</b> - Local server name</li>
+ * <li><b>%D</b> - Time taken to process the request, in millis</li>
+ * <li><b>%T</b> - Time taken to process the request, in seconds</li>
+ * </ul>
+ * <p>In addition, the following aliases for commonly utilized patterns:</p>
+ * <ul>
+ * <li><b>common</b> - <code>%h %l %u %t "%r" %s %b</code> Common Log Format (CLF)</li>
+ * <li><b>combined</b> -
+ * <code>%h %l %u %t "%r" %s %b "%{Referer}i" "%{User-Agent}i"</code> Combined Log Format</li>
+ * </ul>
+ *
+ * @author croudet
+ * @since 2.0
+ */
+public class AccessLogFormatParser {
+    /**
+     * The combined log format.
+     */
+    public static final String COMBINED_LOG_FORMAT = "%h %l %u %t \"%r\" %s %b \"%{Referer}i\" \"%{User-Agent}i\"";
+    /**
+     * The common log format.
+     */
+    public static final String COMMON_LOG_FORMAT = "%h %l %u %t \"%r\" %s %b";
+
+    private List<IndexedLogElement> onRequestElements = new ArrayList<>();
+    private List<IndexedLogElement> onResponseHeadersElements = new ArrayList<>();
+    private List<IndexedLogElement> onResponseWriteElements = new ArrayList<>();
+    private List<IndexedLogElement> onLastResponseWriteElements = new ArrayList<>();
+    private List<IndexedLogElement> constantElements = new ArrayList<>();
```

Comment:
Also please make the fields final if possible

Reply:
Done